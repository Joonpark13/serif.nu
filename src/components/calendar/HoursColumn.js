import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/styles';
import { Typography } from '@material-ui/core';
import {
  columnBorderStyle,
  columnMinHeight,
  dowColumnHeaderHeight,
  cellMinHeight,
  columnHeight,
} from './calendar-constants';

export const styles = {
  hoursColumnContainer: {
    minHeight: columnMinHeight,
    borderRight: columnBorderStyle,
  },
  hoursSpacer: {
    height: dowColumnHeaderHeight,
  },
  hoursColumn: {
    width: 50,
    height: columnHeight,
    display: 'flex',
    flexDirection: 'column',
  },
  hours: {
    flexGrow: 1,
    minHeight: cellMinHeight,
    borderBottom: '1px solid white',
  },
};

/*
 * 0 is 12am
 * 1 - 11 gets 'am'
 * 12 gets 'pm'
 * 13 - 23 gets -12, then 'pm'
 */
export function formatHour(hour) {
  if (hour === 0 || hour === 24) {
    return '12am';
  }
  if (hour <= 11) {
    return `${hour}am`;
  }
  if (hour === 12) {
    return '12pm';
  }
  if (hour <= 23) {
    return `${hour - 12}pm`;
  }
  throw new RangeError('Invalid input. hour must be in the range [0, 24]');
}

function HoursColumn({ hours, classes }) {
  return (
    <div className={classes.hoursColumnContainer}>
      <div className={classes.hoursSpacer} />
      <div className={classes.hoursColumn}>
        {hours.map(hour => (
          <Typography key={hour} align="right" className={classes.hours}>
            {formatHour(hour)}
          </Typography>
        ))}
      </div>
    </div>
  );
}

HoursColumn.propTypes = {
  hours: PropTypes.arrayOf(PropTypes.number).isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export { HoursColumn as UnstyledHoursColumn };

export default withStyles(styles)(HoursColumn);
