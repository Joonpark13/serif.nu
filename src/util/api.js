/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
