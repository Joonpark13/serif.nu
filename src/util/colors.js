export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
