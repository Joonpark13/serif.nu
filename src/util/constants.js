/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
