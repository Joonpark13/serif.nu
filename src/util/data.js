import termsData from 'data/terms.json';

/* eslint-disable import/prefer-default-export */
/* istanbul ignore next */
export const currentTerm = termsData.reduce((highestTerm, term) => {
  if (term.id > highestTerm.id) {
    return term;
  }
  return highestTerm;
});
/* eslint-enable import/prefer-default-export */
