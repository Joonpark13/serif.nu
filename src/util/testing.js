/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
