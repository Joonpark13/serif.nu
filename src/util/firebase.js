import firebase from 'firebase/app';
import 'firebase/firestore';
import 'firebase/auth';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export const db = firebase.firestore();
export const auth = firebase.auth();
