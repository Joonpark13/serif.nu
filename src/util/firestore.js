import firebase from 'firebase/app';
import 'firebase/firestore';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export default firebase.firestore();
