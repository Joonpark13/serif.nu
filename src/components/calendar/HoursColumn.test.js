import React from 'react';
import { shallow } from 'enzyme';
import { mockStyles } from 'util/testing';
import { formatHour, UnstyledHoursColumn, styles } from './HoursColumn';

describe('formatHour', () => {
  it('returns 12am when given 0 or 24', () => {
    expect(formatHour(0)).toBe('12am');
    expect(formatHour(24)).toBe('12am');
  });

  it('returns correctly formatted hour when given an hour between 1 and 11', () => {
    expect(formatHour(7)).toBe('7am');
  });

  it('returns 12pm when given 12', () => {
    expect(formatHour(12)).toBe('12pm');
  });

  it('returns correctly formatted hour when given an hour between 13 and 23', () => {
    expect(formatHour(17)).toBe('5pm');
  });

  it('throws RangeError when given an invalid hour', () => {
    expect(() => { formatHour(25); }).toThrowError(RangeError);
  });
});

describe('HoursColumn', () => {
  it('renders correctly', () => {
    const classes = mockStyles(styles);
    const wrapper = shallow(<UnstyledHoursColumn hours={[1, 2]} classes={classes} />);

    expect(wrapper.get(0)).toMatchSnapshot();
  });
});
