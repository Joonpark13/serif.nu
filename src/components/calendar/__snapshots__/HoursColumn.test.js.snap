// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoursColumn renders correctly 1`] = `
<div
  className=""
>
  <div
    className=""
  />
  <div
    className=""
  >
    <ForwardRef(WithStyles)
      align="right"
      className=""
    >
      1am
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      align="right"
      className=""
    >
      2am
    </ForwardRef(WithStyles)>
  </div>
</div>
`;
